ultralytics/models/yolo/model.py

```diff
@@ -87,18 +89,32 @@ def task_map(self):
             }
         }
 
-    def set_classes(self, classes):
+    def set_classes(self, labels, images=[]):
         """
         Set classes.
 
         Args:
-            classes (List(str)): A list of categories i.e ["person"].
+            labels (List(str)): A list of categories i.e ["person"].
+            images (List()): A list of images.
         """
-        self.model.set_classes(classes)
+
+        classes = labels.copy()
+        for i, img in enumerate(images):
+            if isinstance(img, str):
```

Comment:
Could be interesting to include other types in the future (specifically thinking Numpy or Torch arrays), but fot eh meantime, I think changing this to `isinstance(img, (Path, str))` might be a good addition. Would also be good to make sure that file extension is one of the supported `IMG_FORMATS` from `ultralytics.data.utils`